frontend/src/app/patients/PatientForm.tsx

```diff
@@ -248,16 +252,105 @@ const PatientForm = (props: Props) => {
     } else if (value === allFacilities) {
       value = null;
     }
+    if (errors[name]) {
+      validateField(e);
+    }
     setFormChanged(true);
     setPatient({ ...patient, [e.target.name]: value });
   };
+
+  const validateField = (e: { target: any }, setState = true) => {
```

Comment:
This implementation _works_ but is a little complicated – could you add some docs?